agenta-backend/agenta_backend/services/db_manager.py

```diff
@@ -296,13 +296,41 @@ async def list_apps(**kwargs: dict) -> List[App]:
     if user is None:
         return []
 
-    query_expression = query.eq(AppVariantDB.user_id, user.id) & query.eq(
-        AppVariantDB.is_deleted, False
-    )
-    apps: List[AppVariantDB] = await engine.find(AppVariantDB, query_expression)
-    apps_names = [app.app_name for app in apps]
-    sorted_names = sorted(set(apps_names))
-    return [App(app_name=app_name) for app_name in sorted_names]
+    # Get the organizations the user belongs to
```

Comment:
It seems you just solved the organization problem for list_apps but not the rest of the functions in db_manager. Check `add_variant_based_on_previous` for instance. 

I suggest before implementing the solution, to write a short document describing how each of the api functions would now work after adding the concept of organizations. Basically design how the API would work and why. Then go to implementation. 

The current implementation is a bit strange for instance here. I don't understnad why we check whether user_organizations is None for instacnce